confluent_kafka/src/Producer.c

```diff
@@ -398,6 +398,34 @@ static PyObject *Producer_flush (Handle *self, PyObject *args,
         return cfl_PyInt_FromInt(qlen);
 }
 
+
+static void *Producer_purge (Handle *self, PyObject *args,
+                                 PyObject *kwargs) {
+	int purge_strategy;
+	int blocking = 0;
+	rd_kafka_resp_err_t err;
+        static char *kws[] = { "purge_strategy", "blocking", NULL};
+
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "i|p", kws, &blocking))
+			return NULL;
+	if (blocking==1)
+		err = rd_kafka_purge(self->rk, purge_strategy);
+	else
+		err = rd_kafka_purge(self->rk, purge_strategy|RD_KAFKA_PURGE_F_NON_BLOCKING);
+	if (err == RD_KAFKA_RESP_ERR_NO_ERROR) {
+		Py_RETURN_NONE;
+	}
+	if (err == RD_KAFKA_RESP_ERR__INVALID_ARG) {
```

Comment:
I don't think the specific error code handling is needed here, it is very unlikely to happen, so to keep it simple simply do `cfl_PyErr_format(err, "%s", rd_kafka_err2str(err))`